src/Sarif.Converters/FortifyFprConverter.cs

```diff
@@ -358,22 +393,27 @@ private void ParseLocationFromAnalysisInfo(Result result)
                         }
                     };
 
-                    // Remember the id of the snippet associated with this location.
-                    // We'll use it to fill the snippet text when we read the Snippets element later on.
-                    if (!String.IsNullOrEmpty(snippetId))
+                    // Step past the empty SourceLocation element.
+                    _reader.Read();
+
+                    // If we don't have a label, get the <Action> value
+                    if (string.IsNullOrWhiteSpace(nodeLabel))
                     {
-                        _cflToSnippetIdDictionary.Add(cfl, snippetId);
+                        nodeLabel = _reader.ReadElementContentAsString();
                     }
 
-                    codeFlow.ThreadFlows[0].Locations.Add(cfl);
+                    // Create the region's snippet
+                    physLoc.Region.Snippet = new FileContent
```

Comment:
>Snippet [](start = 35, length = 7)

A snippet contains the source code associated with the region. Is that what you have here? I don't think so. I think you should be populating phyLoc.Annotations here.

btw - Larry and I generally prefer to avoid shortened names, so 'physicalLocation' preferable to 'phyLoc'. :) Hope that's ok with you stylistically

Reply:
The string here is the statement at the codeFlowLocation. Should it be in cfl.Location.Message?

---
In reply to: [207320718](https://github.com/Microsoft/sarif-sdk/pull/981#discussion_r207320718) [](ancestors = 207320718)

Reply:
Put another way: this is the text shown in the call stack nodes in the Fortify client. Our VS addin gets its call tree node text from tfl.Location.PhysicalLocation.Region.Snippet.

---
In reply to: [207381372](https://github.com/Microsoft/sarif-sdk/pull/981#discussion_r207381372) [](ancestors = 207381372,207320718)

Reply:
Fixed, fixed.

---
In reply to: [207320718](https://github.com/Microsoft/sarif-sdk/pull/981#discussion_r207320718) [](ancestors = 207320718)